src/interval.jl

```diff
@@ -187,6 +187,47 @@ Base.isopen(interval::AbstractInterval{T,L,R}) where {T,L,R} = L === Open && R =
 isunbounded(interval::AbstractInterval{T,L,R}) where {T,L,R} = L === Unbounded && R === Unbounded
 isbounded(interval::AbstractInterval{T,L,R}) where {T,L,R} = L !== Unbounded && R !== Unbounded
 
+# Minimum and maximum calls for Closed and Unbounded Intervals, just return first and last.
+function Base.minimum(interval::AbstractInterval{T,L,R}; increment=nothing) where {T,L,R}
+    return first(interval)
+end
+
+function Base.maximum(interval::AbstractInterval{T,L,R}; increment=nothing) where {T,L,R}
+    return last(interval)
+end
+
+# Minimum and maximum calls for open bounds and ambigous types, return open side +- increment
+# Default increment val is eps(T)
+function Base.minimum(interval::AbstractInterval{T,Open,R}; increment=eps(T)) where {T,R}
+    return first(interval) + increment
+end
+
+function Base.maximum(interval::AbstractInterval{T,L,Open}; increment=eps(T)) where {T,L}
+    return last(interval) - increment
+end
```

Comment:
What should I do in the following case:

``` julia
first(interval) + increment > last(interval)
last(interval) - increment < first(interval)
```

should I be returning `nothing` in these cases? Since they are ∉ the interval

Reply:
A very good question. As the provided increment is supposed to be the smallest span between to values I was thinking an exception would be reasonable. Looking at empty intervals (e.g. `Interval{Open,Open}(0, 0)`) should probably be similar to using `minimum`/`maximum` on an empty collection which raises an exception:
```julia
julia> maximum(Int[])
ERROR: ArgumentError: reducing over an empty collection is not allowed
```

Reply:
I suppose it would depend if the other side is open or closed. You would expect an `Interval{T, Closed, Open}` to have a maximum regardless of increment.